sdk/search/search/review/search.api.md

```diff
@@ -131,6 +570,34 @@ export interface RawSearchRequest {
     top?: number;
 }
 
+// @public
+export type RegexFlags = 'CANON_EQ' | 'CASE_INSENSITIVE' | 'COMMENTS' | 'DOTALL' | 'LITERAL' | 'MULTILINE' | 'UNICODE_CASE' | 'UNIX_LINES';
```

Comment:
These can be combined via | in the REST API (e.g.: `"flags": "CANON_EQ | COMMENTS | CASE_INSENSITIVE"`). We'll need a way to model that in the SDK.

Reply:
Hmm seems like we could maybe have it take a `Set<RegexFlags>` and convert it to the appropriate string over the wire

Reply:
@sima-zhu Something like this could work in Java too maybe?

Reply:
I'd say `RegexFlags[]` over `Set<RegexFlags>`. People don't like having to new sets to pass into an API.

Also, in JS, these things are called `RegExp`s not `Regex`es. Worth renaming to align with our builtin type names?

Reply:
Another thought... some of these flags map cleanly to JS RegExp flags (CASE_INSENSITIVE => i, DOTALL => s, MULTILINE => m, UNICODE_CASE* => 'u'). Others don't appear to have a JS equivalent (CANON_EQ, UNIX_LINES, COMMENTS, LITERAL).

We could consider optionally allowing users to specify RegExp flags that have JS equivalents like they would to the RegExp constructor, e.g. "ismu"?

*I'm getting conflicting info on whether Java Regex matches any BMP character without this flag like JS, or just ASCII characters.

Reply:
I do the conversion to/from `RegexFlags[]` now. I didn't rename to `RegExp` since as Brian said, not all map cleanly to JS concepts.